client/app/components/items-list/components/Sidebar.jsx

```diff
@@ -11,15 +11,25 @@ import TagsList from "@/components/TagsList";
  */
 
 export function SearchInput({ placeholder, value, showIcon, onChange }) {
+  const [currentValue, setCurrentValue] = useState(value);
+
+  useEffect(() => {
+    setCurrentValue(value);
+  }, [value]);
+
+  const onInputChange = useCallback(
+    event => {
+      const newValue = event.target.value;
+      setCurrentValue(newValue);
+      onChange(newValue);
+    },
+    [onChange]
+  );
```

Comment:
Why is this needed? It seems that if we set `InputControl`'s `value` as the received `value` prop it will work. Does it create a loop in some cases or something like this?

Reply:
`onChange` is debounced, so if passing `value` prop directly to `Input` component it will update with delay (user will type and see typed text only after they stop typing). This is quite common pattern - use local state to reflect typed text immediately, and sync it with prop when prop changes. Also, `defaultValue` doesn't work in this case because it will be used only when `Input` is mounted and all subsequent updates will be ignored.